build.py

```diff
@@ -1514,6 +1540,36 @@ def add_cpu_libs_to_linux_dockerfile(backends, target_machine):
     return df
 
 
+def change_default_python_version_rhel(version):
+    df = """
+# The python library version available for install via 'yum install python3.X-devel' does not
+# match the version of python inside the RHEL base container. This means that python packages
+# installed within the container will not be picked up by the python backend stub process pybind
+# bindings. It must instead must be installed via pyenv.
+ENV PYENV_ROOT=/tmp/pyenv_build
+RUN curl https://pyenv.run | bash
+ENV PATH="${{PYENV_ROOT}}/bin:$PATH"
+RUN eval "$(pyenv init -)"
+RUN CONFIGURE_OPTS=\"--with-openssl=/usr/lib64\" && pyenv install {} \\
+    && cp ${{PYENV_ROOT}}/versions/{}/lib/libpython3* /usr/lib64/""".format(
+        version, version
+    )
+    df += """
+# RHEL image has several python versions. It's important
+# to set the correct version, otherwise, packages that are
+# pip installed will not be found during testing.
+ENV PYVER {}
+ENV PYTHONPATH /opt/python/v
+RUN ln -sf ${{PYENV_ROOT}}/versions/${{PYVER}}* ${{PYTHONPATH}}
+ENV PYBIN ${{PYTHONPATH}}/bin
+ENV PYTHON_BIN_PATH ${{PYBIN}}/python${{PYVER}}
+ENV PATH ${{PYBIN}}:${{PATH}}
```

Comment:
Hard to say how big is our dockerfile in terms on lines trying to make sure it compact.
```suggestion
ENV PYBIN=${{PYTHONPATH}}/bin PYTHON_BIN_PATH=${{PYBIN}}/python${{PYVER}} PATH=${{PYBIN}}:${{PATH}}
```

Reply:
Wondering why did you use a a notation `${{var}}` vs `${var}`

Reply:
Because this code is written in a Python script that uses `format`, we need to escape the `{}` otherwise it'll interpret it as a variable instead of treating it like a literal like we want.

Context: https://stackoverflow.com/questions/5466451/how-do-i-escape-curly-brace-characters-characters-in-a-string-while-using

Reply:
Local testing demonstrates that we cannot define a variable using another variable that is defined in the same line. For example, `PYTHON_BIN_PATH` and `PATH` here both depend on the definition of `PYBIN`. It must be defined before we use it to set these dependent values.